test/api/v1/client_test.go

```diff
@@ -165,23 +174,34 @@ func TestClientInstanceMethods(t *testing.T) {
 	err = suite.APIClient.DeleteInstance(suite.Context(), deleteRequest)
```

Comment:
can move this after the Retry to ensure the no-op occurs after the previous attempt was successful? Otherwise we might be doing it too quickly and the underlying instances are still not terminated.